### ISSUE TITLE
Build failure with 0.99.8 / 3.0.0

### ISSUE THREAD

**Issue by brad0** [clangen/musikcube#606]
Trying to update the OpenBSD port from 0.98.x to 0.99.x / 3.0.0 I ran into the following build failure..

```
/home/brad/tmp/ffmpeg-ports/ports/pobj/musikcube-0.99.8/bin/c++ -DASIO_STANDALONE -DBOOST_DATE_TIME_NO_LIB -DBOOST_REGEX_NO_LIB -DMCSDK_DEFINE_EXPORTS -DNO_NCURSESW -D_WEBSOCKETPP_CPP11_RANDOM_DEVICE_ -D_WEBSOCKETPP_CPP11_TYPE_TRAITS_ -Dmusikcore_EXPORTS -I/usr/local/include -I/usr/local/opt/openssl/include -I/usr/local/opt/ncurses/include -I/home/brad/tmp/ffmpeg-ports/ports/pobj/musikcube-0.99.8/musikcube-0.99.8/src -I/home/brad/tmp/ffmpeg-ports/ports/pobj/musikcube-0.99.8/musikcube-0.99.8/src/musikcore -I/home/brad/tmp/ffmpeg-ports/ports/pobj/musikcube-0.99.8/musikcube-0.99.8/src/musikcube -I/home/brad/tmp/ffmpeg-ports/ports/pobj/musikcube-0.99.8/musikcube-0.99.8/src/musikcube/cursespp -I/home/brad/tmp/ffmpeg-ports/ports/pobj/musikcube-0.99.8/musikcube-0.99.8/src/3rdparty/include -I/home/brad/tmp/ffmpeg-ports/ports/pobj/musikcube-0.99.8/musikcube-0.99.8/src/3rdparty/asio/asio/include -I/home/brad/tmp/ffmpeg-ports/ports/pobj/musikcube-0.99.8/musikcube-0.99.8/src/musikcore/. -I/home/brad/tmp/ffmpeg-ports/ports/pobj/musikcube-0.99.8/musikcube-0.99.8/src/musikcore/.. -I/home/brad/tmp/ffmpeg-ports/ports/pobj/musikcube-0.99.8/musikcube-0.99.8/src/musikcore/../3rdparty/include -I/home/brad/tmp/ffmpeg-ports/ports/pobj/musikcube-0.99.8/musikcube-0.99.8/src/musikcore/../3rdparty/include/md5 -I/home/brad/tmp/ffmpeg-ports/ports/pobj/musikcube-0.99.8/musikcube-0.99.8/src/musikcore/../3rdparty/include/sqlite -O2 -pipe -std=c++17 -Wno-unused-result -Wno-deprecated-declarations -DNDEBUG -std=c++17 -O2 -frtti -fexceptions -Wno-psabi -fPIC -MD -MT src/musikcore/CMakeFiles/musikcore.dir/c_context.cpp.o -MF src/musikcore/CMakeFiles/musikcore.dir/c_context.cpp.o.d -o src/musikcore/CMakeFiles/musikcore.dir/c_context.cpp.o -c /home/brad/tmp/ffmpeg-ports/ports/pobj/musikcube-0.99.8/musikcube-0.99.8/src/musikcore/c_context.cpp
In file included from /home/brad/tmp/ffmpeg-ports/ports/pobj/musikcube-0.99.8/musikcube-0.99.8/src/musikcore/c_context.cpp:35:
In file included from /home/brad/tmp/ffmpeg-ports/ports/pobj/musikcube-0.99.8/musikcube-0.99.8/src/musikcore/pch.hpp:48:
In file included from /home/brad/tmp/ffmpeg-ports/ports/pobj/musikcube-0.99.8/musikcube-0.99.8/src/3rdparty/include/websocketpp/config/asio_no_tls_client.hpp:32:
In file included from /home/brad/tmp/ffmpeg-ports/ports/pobj/musikcube-0.99.8/musikcube-0.99.8/src/3rdparty/include/websocketpp/transport/asio/endpoint.hpp:32:
In file included from /home/brad/tmp/ffmpeg-ports/ports/pobj/musikcube-0.99.8/musikcube-0.99.8/src/3rdparty/include/websocketpp/transport/asio/connection.hpp:31:
In file included from /home/brad/tmp/ffmpeg-ports/ports/pobj/musikcube-0.99.8/musikcube-0.99.8/src/3rdparty/include/websocketpp/transport/asio/base.hpp:31:
/home/brad/tmp/ffmpeg-ports/ports/pobj/musikcube-0.99.8/musikcube-0.99.8/src/3rdparty/include/websocketpp/common/asio.hpp:46:14: fatal error: 'asio/version.hpp' file not found
    #include <asio/version.hpp>
             ^~~~~~~~~~~~~~~~~~
1 error generated.
```

**Comment by clangen** [comment]
@brad0 thanks for the report. Newer versions of `musikcube` include `asio` as a git submodule -- fixing this issue should be as easy as ensuring all submodules are fetched before building. Alternatively, adding `asio` as a dependency would fix the issue, if OpenBSD has a port available for that project.

**Comment by brad0** [comment]
Even when I tried grabbing the tar archive from GitHub it seemed to be missing the ```asio``` submodule. Our port currently pulls from the repo so it is expected that the submodule is not initialized. Anyway, we do not have an ```asio``` port at the moment. I took a peek at FreeBSD and NetBSD's ports and it is pretty easy, being header-only. So I whipped togethera port in 5 mins.

```musikcube``` 3.0.0 is currently building well past the initial failure point.
